src/Diagnostics/DiagnosticDispatcher.php

```diff
@@ -0,0 +1,133 @@
+<?php declare(strict_types = 1);
+
+namespace Contributte\EventDispatcher\Diagnostics;
+
+use Contributte\EventDispatcher\Tracy\Panel;
+use Nette\Utils\ObjectHelpers;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+class DiagnosticDispatcher implements EventDispatcherInterface
+{
+
+	/** @var EventDispatcherInterface */
+	private $original;
+
+	/** @var Panel|null */
+	private $panel = null;
+
+	/** @var LoggerInterface|null */
+	private $logger = null;
+
+	public function __construct(EventDispatcherInterface $original)
+	{
+		$this->original = $original;
+	}
+
+	public function setPanel(Panel $panel): void
```

Comment:
What it has connection to panel?

Reply:
Panel was different from loggers, now simplified and this reference removed